plugins/modules/zos_zfs_resize.py

```diff
@@ -0,0 +1,660 @@
+    size: 20
+
+- name: Resize an aggregate data set to 4 Megabytes.
+  zos_zfs_resize:
+    target: TEST.ZFS.DATA
+    space_type: m
+    size: 4
+
+- name: Resize an aggregate data set to 1000 Kilobytes and no auto increment if is shrinking.
+  zos_zfs_resize:
+    target: TEST.ZFS.DATA
+    size: 1000
+    no_auto_increment: True
+
+- name: Resize an aggregate data set and get verbose output.
+  zos_zfs_resize:
+    target: TEST.ZFS.DATA
+    size: 2500
+    verbose: True
+
+- name: Resize an aggregate data set and get the full trace on an uss file.
+  zos_zfs_resize:
+    target: TEST.ZFS.DATA
+    size: 2500
+    trace_destination: /tmp/helper.txt
+
+- name: Resize an aggregate data set and get the full trace on a member of pds.
+  zos_zfs_resize:
+    target: TEST.ZFS.DATA
+    size: 2500
+    trace_destination: "TEMP.HELPER.STORAGE(RESIZE)"
+
+- name: Resize an aggregate data set and get the full trace on an uss file and verbose output.
+  zos_zfs_resize:
+    target: TEST.ZFS.DATA
+    size: 2500
+    verbose: True
+    trace_destination: /tmp/helper.txt
+"""
+
+RETURN = r"""
+cmd:
+    description: The zfsadm command executed on the remote node.
+    returned: always
+    type: str
+    sample: zfsadm grow -aggregate SOMEUSER.VVV.ZFS -size 4096
+target:
+    description: The Fully Qualified Name of the resized zfs data set.
+    returned: always
+    type: str
+    sample: SOMEUSER.VVV.ZFS
+mount_target:
+    description: The original share/mount.
+    returned: always
+    type: str
+    sample: /tmp/zfs_agg
+size:
+    description: The approximate size, of the data set after the resizing is performed.
+    returned: always
+    type: int
+    sample: 4024
+rc:
+    description: The return code of the zfsadm command.
+    returned: always
+    type: int
+    sample: 0
+old_size:
+    description: The reported size, in space_type, of the data set before the resizing is performed.
+    returned: always
+    type: float
+    sample: 3096
+old_free_space:
+    description: The reported size, in space_type, of the free space in the data set before the resizing is performed.
+    returned: always
+    type: float
+    sample: 108
+new_size:
+    description: The reported size, in space_type, of the data set after the resizing is performed.
+    returned: success
+    type: float
+    sample: 4032
+new_free_space:
+    description: The reported size, in space_type, of the free space in the data set after the resizing is performed.
+    returned: success
+    type: float
+    sample: 48
+space_type:
+    description: The reported space_type that the size and free variables are reported.
+    returned: always
+    type: str
+    sample: k
+verbose_output:
+    description: If C(verbose=true) the full traceback of operation will show on this variable. If C(trace) will return the data set or path name.
+    returned: C(verbose=true) and success
+    type: str
+    sample: 6FB2F8 print_trace_table printing contents of table Main Trace Table
```

Comment:
```suggestion
    sample: 6FB2F8 print_trace_table printing contents of table Main Trace Table...
```

The addition of the periods is just to indicate that the text will likely be longer than what you're including.